File: WebAPI/StoryBoard/DTO/ProjectDto.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DTO;

public class ProjectDto
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string UrlImage { get; set; }
    public DateTime CreationDate { get; set; }
}

File: UI/StoryBoard/src/app/dtos/project-dto/project-dto.ts
export class ProjectDto {
    id : number = 0;
    name : string = "";
    urlImage : string = "";
    creationDate : Date = new Date;
    description : string = "";

    static createFromFormValues(formValues: any): ProjectDto {
        const project = new ProjectDto();
        project.name = formValues.name;
        project.urlImage = formValues.urlImage;
        project.creationDate = new Date;
        project.description = formValues.description;
        return project;
    }
}